notebooks/linear_regression.livemd

```diff
@@ -0,0 +1,610 @@
+# Linear Regression in Practice
+
+```elixir
+Mix.install(
+  [
+    {:scholar, github: "elixir-nx/scholar"},
+    {:explorer, github: "elixir-nx/explorer"},
+    {:nx, github: "elixir-nx/nx", sparse: "nx", override: true},
+    {:exla, github: "elixir-nx/nx", sparse: "exla", override: true},
+    {:rustler, ">= 0.0.0", optional: true},
+    {:req, "~> 0.3.3"},
+    {:vega_lite, "~> 0.1.6"},
+    {:kino_vega_lite, "~> 0.1.7"},
+    {:kino, "~> 0.8.0"}
+  ],
+  config: [
+    nx: [default_defn_options: [compiler: EXLA]]
+  ]
+)
+```
+
+## Introduction
+
+In the livebook, we will cover the typical use cases of linear regression on practical examples.
+
+```elixir
+alias VegaLite, as: Vl
+require Explorer.DataFrame
+```
+
+<!-- livebook:{"branch_parent_index":0} -->
+
+## Linear Regression on Synthetic Data
+
+```elixir
+use Nx.Defn.Kernel
+```
+
+Before we dive into real-life use cases of linear regression, we start with a simpler one. We will generate data with a linear pattern and then use `Scholar.Linear.LinearRegression` to compute regression.
+
+Firstly, we generate the data which simulates the function $f(x) = 3x + 4$ with added uniform, zero-mean noise. `Nx.Random.uniform` creates a tensor with a given shape and type.
+
+```elixir
+key = Nx.Random.key(42)
+size = 100
+{uniform_x, new_key} = Nx.Random.uniform(key, shape: {size, 1}, type: :f32)
+{uniform_y, _} = Nx.Random.uniform(new_key, shape: {size, 1}, type: :f32)
+x = 2 * uniform_x
+y = 3 * x + 4 + (uniform_y - Nx.mean(uniform_y))
```

Comment:
```suggestion
{x, new_key} = Nx.Random.uniform(key, 0, 2shape: {size, 1}, type: :f32)
{noise, _} = Nx.Random.uniform(new_key, -0.5, 0.5, shape: {size, 1}, type: :f32)
y = 3 * x + 4 + noise
```

Reply:
In line 46 there is no comma between 0 and 2, I will add this change separately.